src/internal/connector/discovery/api/users.go

```diff
@@ -59,13 +60,21 @@ const (
 // https://stackoverflow.com/questions/64044266/error-message-unsupported-or-invalid-query-filter-clause-specified-for-property
 //
 //nolint:lll
-var userFilterNoGuests = "onPremisesSyncEnabled eq true OR userType eq 'Member'"
+var userFilterNoGuests = "onPremisesSyncEnabled eq true OR userType ne 'Guest'"
```

Comment:
`onPremisesSyncEnabled` this was `null` for their users as well I think. Are we sure this is something we want?

Reply:
Unclear - I can't find any clear documentation on whether we can have "guest on-prem users" - that would be the concern with this check.

If we remove it - run the risk of not discovering all users
If we keep it - and the check is incorrect - will discover users but backup will probably fail with a "no drive" or "no mailbox found".

Leaning towards the latter for now i.e. keep it.

Reply:
Fine by me